components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/model/OpenAPIRequest.java

```diff
@@ -72,8 +87,12 @@ public OpenAPIRequest(MessageContext messageContext) {
                 .stream()
                 .collect(Collectors.toMap(Map.Entry::getKey, entry -> Collections.singleton(entry.getValue())));
         //Set transport headers
+        String contentTypeHeader = "content-type";
         for (Map.Entry<String, Collection<String>> header : headerMap.entrySet()) {
-            headers.put(header.getKey(), header.getValue().iterator().next());
+            String headerKey = header.getKey();
+            String value =  header.getValue().iterator().next();
+            headerKey = headerKey.toLowerCase().equals(contentTypeHeader) ? "Content-Type" : headerKey;
```

Comment:
equalignorecase is better isn't it?

Reply:
since comparing the lowercase values only, it should be fine.

Reply:
I mean instead of toLowerCase().equals() we can use equalignorecase()

Reply:
Fixed with b472b5ce30c..3a94da1e653